### PR TITLE
Refactor deep dives a little for more nesting.

### PR DESCRIPTION
**Pull request by jezdez**
### Description

This is just a tiny deep dive refactor to nest them and prepare to add additional ones in the future.

@jaimergp Should we add the tech spec into a new `specs` directory as well (since it's currently orphaned)?

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [ ] ~Add / update necessary tests?~
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by jaimergp** [comment]
Yes, good idea. Do you want me to push it directly to this branch?

**Review by dholth — DISMISSED** [review]
thanks for organizing the docs

File: docs/source/dev-guide/deep-dives/index.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Deep dives
+
+This section contains a series of deep dives into particular gnarly parts
```
Comment (kathatherine) [inline docs/source/dev-guide/deep-dives/index.md:3]: Particularly gnarly?

Or did you mean "particular" as in "various"?

**Review by kathatherine — CHANGES_REQUESTED** [review]
One question about wording, but looks good to me otherwise!

File: docs/source/dev-guide/deep-dives/index.md (hunk 1 above)
Comment (kathatherine) [reply docs/source/dev-guide/deep-dives/index.md:3]: No one knows

Comment (dholth) [reply docs/source/dev-guide/deep-dives/index.md:3]: Here are some other cool words. https://www.mariowiki.com/Special_Zone

I would suggest being boring and replacing gnarly with complex or some other word; perhaps the reader does not agree that the part is gnarly.

Comment (kathatherine) [reply docs/source/dev-guide/deep-dives/index.md:3]: That's a fair point. We also want to keep in mind Google translate and whether or not "gnarly" would translate well out of English. Likely not. :) Maybe "complex" would be better.

Comment (jezdez) [reply docs/source/dev-guide/deep-dives/index.md]: Thank you both, this is really appreciated!

File: docs/source/dev-guide/deep-dives/index.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Deep dives
+
+This section contains a series of deep dives into particular complex parts
```
Comment (travishathaway) [inline docs/source/dev-guide/deep-dives/index.md:3]:
```suggestion
This section contains a series of deep dives into particularly complex parts
```

File: docs/source/dev-guide/specs/index.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Specifications
+
+This section contains an incomplete list of conda specifications that
+may or may not be related to [Conda Enhancement Proposal](https://github.com/conda-incubator/ceps).
```
Comment (travishathaway) [inline docs/source/dev-guide/specs/index.md:4]:
```suggestion
may or may not be related to [Conda Enhancement Proposals](https://github.com/conda-incubator/ceps).
```

I think this is plural, right? Feel free to ignore if not.

File: docs/source/dev-guide/specs/solver-state.md [hunk 4]
```diff
@@ -1,18 +1,16 @@
...
-
 # Technical specification: solver state
 
-> Note: This document is a technical specification, which might not be the best way to learn about
-> how the solver works. For that, refer to {ref}`deep_dive_install` and {ref}`deep_dive_solvers`.
+:::{admonition} Note
+:class: warning
+
+This document is a technical specification, which might not be the best way to learn about
+how the solver works. For that, refer to {doc}`deep-dives/install` and
```
Comment (travishathaway) [inline docs/source/dev-guide/specs/solver-state.md:11]: This link is broken on the preview ☹️ https://continuumio-conda-docs--12041.com.readthedocs.build/projects/conda/en/12041/dev-guide/specs/solver-state.html

Not sure how to fix it. Probably just a MD vs. RST thing.

**Review by travishathaway — CHANGES_REQUESTED** [review]
Found at least one broken link. A couple word issues.

File: docs/source/dev-guide/specs/solver-state.md (hunk 4 above)
Comment (jezdez) [reply docs/source/dev-guide/specs/solver-state.md:11]: Huh, weird! That works for me locally. Looking into it..

**Review by kathatherine — APPROVED** [review]
Links look good to me in the current branch build. 👍

File: docs/source/dev-guide/deep-dives/install.md [hunk 5]
```diff
@@ -6,8 +6,6 @@ Some resources that helped draft this document:
 * https://www.anaconda.com/blog/understanding-and-improving-condas-performance
 -->
 
-(deep_dive_install)=
-
 # Deep dive: conda install
```
Comment (kenodegard) [inline docs/source/dev-guide/deep-dives/install.md:9]: Let's remove this title redundancy:

Screen Shot 2022-11-14 at 10 25 39 AM


```suggestion
# `conda install`
```

File: docs/source/dev-guide/deep-dives/solvers.md [hunk 6]
```diff
@@ -1,8 +1,6 @@
-(deep_dive_solvers)=
-
 # Deep dive: solvers
```
Comment (kenodegard) [inline docs/source/dev-guide/deep-dives/solvers.md:1]:
```suggestion
# Solvers
```

**Review by kenodegard — CHANGES_REQUESTED** [review]
Minor title changes. Similar edits are needed for the other deep-dive docs.

**Review by kenodegard — APPROVED** [review]

**Pull request merged**